framework/filters/AccessRule.php

```diff
@@ -108,7 +122,7 @@ class AccessRule extends Component
     public function allows($action, $user, $request)
     {
         if ($this->matchAction($action)
-            && $this->matchRole($user)
+            && $this->matchRole($user, $action)
```

Comment:
although it suits your usecase, this is a code smell. It adds confusion as $action already had a match with matchAction(). IMO, any further complexification of matching should be done in a cleaner OO way. See next comment.